docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,328 @@
+- Annotate a multi-arch image using OCI image index: `oras manifest create/update --annotation`
+- List platform information of tags in a repository: `oras repo tags`
+
+The proposal uses an OCI image layout as the local store to create a multi-arch image as an OCI image index locally, then push the multi-arch image to the remote registry with ORAS.  
+
+### User scenario and desired experience
+
+The sample workflow will be as following:
+
+1. Assume there are two arch-specific images tagged as _“v1-linux-amd64”_ and _“v1-linux-arm64” and have been pushed to an OCI image layout _layout-dir_. List the tags in the OCI image layout:
+
+```bash
+$ oras repo tags --oci-layout layout-dir 
+
+v1-linux-amd64 
+v1-linux-arm64 
+```
+
+2. Create a multi-arch image by composing two image manifests into an image index, tag it with “v1” and push the image index to an OCI image layout “layout-dir” automatically:  
+
+```
+$ oras manifest index create --oci-layout layout-dir:v1 v1-linux-amd64 v1-linux-arm64 
+
+Resolving  v1-linux-amd64 
+Resolved   v1-linux-amd64 
+Resolving  v1-linux-arm64 
+Resolved   v1-linux-arm64 
+Packed     edb5bc1f0b5c application/vnd.oci.image.index.v1+json 
+Pushed     [oci-layout] layout-dir:v1 
+Digest: sha256:edb5bc1f0b5c21e9321b34e50c92beae739250fb8840905 
+
+An image index has been created and pushed to layout-dir:v1 
+```
+
+3. Inspect the image index from the OCI image layout: 
+
+```bash
+$ oras manifest fetch --oci-layout layout-dir:v1 --pretty  
+
+{ 
+  "schemaVersion": 2, 
+  "mediaType": "application/vnd.oci.image.index.v1+json", 
+  "manifests": [ 
+    { 
+      "mediaType": "application/vnd.oci.image.manifest.v1+json", 
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a", 
+      "size": 1239, 
+      "platform": { 
+        "architecture": "amd64", 
+        "os": "linux" 
+      } 
+    }, 
+    { 
+      "mediaType": "application/vnd.oci.image.manifest.v1+json", 
+      "digest": "sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a", 
+      "size": 1239, 
+      "platform": { 
+        "architecture": "arm64", 
+        "os": "linux" 
+      } 
+    } 
+  ] 
+} 
+```
+
+4. Add (or change) annotations to an image index in the OCI layout: 
+
+```bash
+$ oras manifest index update --oci-layout layout-dir:v1 --annotation “platform=multi-arch” 
+
+Resolving  v1 
+Resolved   sha256:2af402374d4c9297bf077b1e722d523c5ce37256d1d04 
+Updating anootation  v1 
+Updated    sha256:447c006400aae8b0d302cafba172f5da9889d50af085a 
+Pushed     [oci-layout] layout-dir:v1 
+Digest: sha256:447c006400aae8b0d302cafba172f5da9889d50af085a 
+Updated annotations of layout-dir:v1 
+```
+
+5. Attach an annotation as a referrer to an existing multi-arch image and its child arch-specific image manifest (without mutating its digest) in one shot. The same annotation is supposed to be added to the parent image index and its child arch-specific image manifest recursively: 
```

Comment:
In favor of extending attach for annotations.

Reply:
This will be follow-up enhancements in the next release.

See the issue: Support attach annotations to image index and its child image manifest in `oras attach`: https://github.com/oras-project/oras/issues/1531